feat(tui): clarify Windows auto mode requirements

joshka-oai opened the pull request:
## Summary
  - Coerce Windows `workspace-write` configs back to read-only, surface the forced downgrade in the approvals popup,
  and funnel users toward WSL or Full Access.
  - Add WSL installation instructions to the Auto preset on Windows while keeping the preset available for other
  platforms.
  - Skip the trust-on-first-run prompt on native Windows so new folders remain read-only without additional
  confirmation.
  - Expose a structured sandbox policy resolution from config to flag Windows downgrades and adjust tests (core,
  exec, TUI) to reflect the new behavior; provide a Windows-only approvals snapshot.

  ## Testing
  - cargo fmt
  - cargo test -p codex-core config::tests::add_dir_override_extends_workspace_writable_roots
  - cargo test -p codex-exec suite::resume::exec_resume_preserves_cli_configuration_overrides
  - cargo test -p codex-tui chatwidget::tests::approvals_selection_popup_snapshot
  - cargo test -p codex-tui approvals_popup_includes_wsl_note_for_auto_mode
  - cargo test -p codex-tui windows_skips_trust_prompt
  - just fix -p codex-core
  - just fix -p codex-tui

joshka-oai committed:
feat(tui): clarify Windows auto mode requirements

Warn native Windows users that the Auto preset requires WSL and show the WSL install steps in a dedicated popup.

joshka-oai committed:
tweak message

dylan-hurd-oai reviewed (commented):
Want to clarify the default approval mode!

joshka-oai commented on codex-rs/tui/src/chatwidget.rs:
```diff
@@ -1804,7 +1806,14 @@ impl ChatWidget {
             let is_current =
                 current_approval == preset.approval && current_sandbox == preset.sandbox;
```
I initially ran codex web against this with 4x. There seemed to be a couple of approaches that it came up with for setting the defaults properly, and I wanted to check them on a proper windows setup to understand each properly. So I just put the messaging side on this PR for now as it's mostly orthogonal.

dylan-hurd-oai commented:
CI failures seem unrelated? I restarted the failed jobs

joshka-oai committed:
feat(windows): clarify forced read-only sandbox

- expose a structured sandbox policy resolution that flags Windows downgrades\n- show a banner in the approvals popup directing users to WSL or Full Access

joshka-oai committed:
fix(windows): skip trust prompt when auto mode forced off

joshka-oai committed:
fix(windows): skip trust setup when auto mode unavailable

joshka-oai commented:
Updated flow including the proper read-only downgrade:

https://github.com/user-attachments/assets/26f74888-0a7f-40fc-8f26-21fd3c40789d

joshka-oai committed:
test(windows): align forced read-only expectations

Update Windows-only assertions to accept the sandbox downgrade and regenerate the approvals popup snapshot to include the WSL hint.

joshka-oai committed:
test(tui): add Windows approvals snapshot

dylan-hurd-oai reviewed (approved):
Some small copy stuff but code LGTM

joshka-oai committed:
use sentence case on approval preset descriptinos

joshka-oai committed:
Change text to read re-enable auto mode

joshka-oai committed:
Fix tests broken by merge
